tests/merch.order.test.ts

```diff
@@ -0,0 +1,721 @@
+
+    await new PortalState()
+      .createUsers(member)
+      .createMerchItemOption(affordableOption1)
+      .createMerchItemOption(affordableOption2)
+      .createOrderPickupEvents(pickupEvent)
+      .write();
+
+    const emailService = mock(EmailService);
+    when(emailService.sendOrderConfirmation(member.email, member.firstName, anything()))
+      .thenResolve();
+    const order = [
+      {
+        option: affordableOption1.uuid,
+        quantity: 1,
+      },
+      {
+        option: affordableOption2.uuid,
+        quantity: 1,
+      },
+    ];
+    const placeMerchOrderRequest = {
+      order,
+      pickupEvent: pickupEvent.uuid,
+    };
+    const placeMerchOrderResponse = await ControllerFactory
+      .merchStore(conn, instance(emailService))
+      .placeMerchOrder(placeMerchOrderRequest, member);
+    const placedOrder = placeMerchOrderResponse.order;
+
+    expect(placedOrder.items).toHaveLength(2);
+    expect(placedOrder.status).toStrictEqual(OrderStatus.PLACED);
+    expect(member.credits).toEqual(5000);
+    verify(emailService.sendOrderConfirmation(member.email, member.firstName, anything()))
+      .called();
+  });
+
+  test('members can cancel orders that they\'ve placed and receive a full refund to their order', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake({ credits: 10000 });
+    const affordableOption = MerchFactory.fakeOption({
+      quantity: 1,
+      price: 2000,
+      discountPercentage: 0,
+    });
+    const pickupEvent = MerchFactory.fakeFutureOrderPickupEvent();
+
+    await new PortalState()
+      .createUsers(member)
+      .createMerchItemOption(affordableOption)
+      .createOrderPickupEvents(pickupEvent)
+      .write();
+
+    const emailService = mock(EmailService);
+    when(emailService.sendOrderConfirmation(member.email, member.firstName, anything()))
+      .thenResolve();
+    when(emailService.sendOrderCancellation(member.email, member.firstName, anything()))
+      .thenResolve();
+
+    // place order
+    const order = [
+      {
+        option: affordableOption.uuid,
+        quantity: 1,
+      },
+    ];
+    const placeMerchOrderRequest = {
+      order,
+      pickupEvent: pickupEvent.uuid,
+    };
+    const merchController = ControllerFactory.merchStore(conn, instance(emailService));
+    const placedOrderResponse = await merchController.placeMerchOrder(placeMerchOrderRequest, member);
+
+    // cancel order
+    const { uuid } = placedOrderResponse.order;
+    await merchController.cancelMerchOrder({ uuid }, member);
+
+    // get order, making sure state was updated and user has been refunded
+    const cancelledOrderResponse = await merchController.getOneMerchOrder({ uuid }, member);
+    const cancelledOrder = cancelledOrderResponse.order;
+
+    expect(cancelledOrder.status).toEqual(OrderStatus.CANCELLED);
+    expect(member.credits).toEqual(10000);
```

Comment:
When `member` is passed into the controller call, its credits manually get updated in the activity repository by a `user.credits += credits` call, so the object does get modified. In the cases where `member` is never passed into a controller (like in `cancelAllPendingOrders`), then we can manually reload with `await member.reload()` - I think I'll  add this call in every test though so that it's clearer